crates/agglayer-contracts/src/settler.rs

```diff
@@ -1,69 +1,99 @@
+            // If specific error decoding fails, try to extract a revert reason string
+            if let Some(reason) = alloy::sol_types::decode_revert_reason(revert_data.as_ref()) {
+                return Some(reason);
+            }
 
-    fn build_pending_transaction(
-        &self,
-        tx_hash: H256,
-    ) -> PendingTransaction<'_, <Self::M as Middleware>::Provider> {
-        PendingTransaction::new(tx_hash, self.rpc.as_ref().provider())
-    }
+            // Fall back to hex representation of revert data
+            return Some(format!("0x{}", hex::encode(revert_data)));
+        }
 
-    fn decode_contract_revert(error: &ContractError<Self::M>) -> Option<String> {
-        error
-            .decode_contract_revert::<PolygonRollupManagerErrors>()
-            .map(|err| format!("{err:?}"))
+        // For non-revert errors, return the debug representation
+        Some(format!("{error:?}"))
     }
 
-    fn build_verify_pessimistic_trusted_aggregator_call(
+    async fn verify_pessimistic_trusted_aggregator(
         &self,
         rollup_id: u32,
         l_1_info_tree_leaf_count: u32,
         new_local_exit_root: [u8; 32],
         new_pessimistic_root: [u8; 32],
-        proof: ::ethers::core::types::Bytes,
-        custom_chain_data: ::ethers::core::types::Bytes,
-    ) -> ContractCall<Self::M, ()> {
-        self.inner.verify_pessimistic_trusted_aggregator(
+        proof: Bytes,
+        custom_chain_data: Bytes,
+    ) -> Result<PendingTransactionBuilder<alloy::network::Ethereum>, ContractError> {
+        // Build the transaction call
+        let mut tx_call = self.inner.verifyPessimisticTrustedAggregator(
             rollup_id,
             l_1_info_tree_leaf_count,
-            new_local_exit_root,
-            new_pessimistic_root,
+            new_local_exit_root.into(),
+            new_pessimistic_root.into(),
             proof,
             custom_chain_data,
+        );
+
+        // Check if the low gas fail point is active and set the low gas if it is
+        #[cfg(feature = "testutils")]
+        if fail::eval(
+            "notifier::packer::settle_certificate::gas_estimate::low_gas",
+            |_| true,
         )
+        .unwrap_or(false)
+        {
+            // Set deliberately low gas to cause failure
+            warn!(
+                "FAIL POINT ACTIVE: low gas fail point active for rollup_id: {}",
+                rollup_id
+            );
+            tx_call = tx_call.gas(30000);
+        }
+
+        // Check if a gas multiplier factor is provided
+        if self.gas_multiplier_factor != 100 {
+            // Apply gas multiplier if it's not the default (100).
+            // First estimate gas, then multiply by the factor.
+            let gas_estimate = tx_call.estimate_gas().await?;
+            let adjusted_gas =
+                (gas_estimate.saturating_mul(self.gas_multiplier_factor as u64)) / 100;
+            warn!(
```

Comment:
Seems a warning every time a certificate is about to be settled with a gas adjustment in force seems a bit excessive. An `info` or even `debug` seems good enough? Is there a reason for this being a warning?

Reply:
Ok, I'll remove warning